www/js/pedagogical_module.js

```diff
@@ -0,0 +1,217 @@
+ * Pedagogical Module class used to solve Dragoon problems
+ * @author: Brandon Strong
+ **/
+
+/* global define */
+
+define([
+	"dojo/_base/array", "dojo/_base/declare", "dojo/_base/lang", "./equation", "dojo/dom"
+], function(array, declare, lang, check, dom){
+	// Summary: 
+	//			Processes student selections and returns instructions to the 
+	//			program
+	// Description:
+	//			A pedagogical module that accepts student entries, and returns 
+	//			an object with the ID of the node, a message with encouragement 
+	//			or a hint, and the status of the attempt (correct, incorrect, 
+	//			demo, or premature).
+	// Tags:
+	//			pedagogical module (PM), student mode, coached mode
+
+	var descriptionTable = {
+		// Summary: This table is used for determining the proper response to a student's 'description' answer (see 
+		//		'Pedagogical_Module.docx' in the documentation)
+		correct: {
+			feedback: function(obj, part){
+				state(obj, part, "correct");
+				message(obj, part, "correct");
+				disable(obj, part, true);
+				//disable(obj, "type", false);
+			}
+		},
+		incorrect: {
+			feedback: function(obj, part){
+				state(obj, part, "incorrect");
+				message(obj, part, "incorrect");
+			}
+		}
+	};
+
+	var nodeEditorActionTable = {
+		// Summary: This table is used for determining the proper response to a student's answers in the 
+		//		remaining sections (see 'Pedagogical_Module.docx' in the documentation)
+
+		//		Node Editor action table will be used for any activity that uses existing node editor.
+		//		All the actions remain same, only add additional field(s) in _getInterpretation and _enableNext
+		correct: {
+			feedback: function(obj, part){
+				state(obj, part, "correct");
+				message(obj, part, "correct");
+				disable(obj, part, true);
+			}
+		},
+		incorrect: {
+			feedback: function(obj, part){
+				state(obj, part, "incorrect");
+			}
+		}
+	};
+	
+	/*
+	 * Add additional tables for activities that does not use node editor.
+	 */
+	//Declare variable for accessing state.js module
+	var record = null;
+
+	/*****
+	 * Summary: The following four functions are used by the above tables to push
+	 *		statuses and messages to the return object array.
+	 *****/
+	function state(/*object*/ obj, /*string*/ nodePart, /*string*/ status){
+		obj.push({id: nodePart, attribute: "status", value: status});
+		if(status==="premature"){
+			obj.push({id: nodePart, attribute: "value", value: ""});
+		}
+	}
+
+	function message(/*object*/ obj, /*string*/ nodePart, /*string*/ status){
+		// TO DO : Add Hint messages
+		obj.push({id: "message", attribute: "append", value: "The value entered for the " + nodePart + " is " + status + "."});
+	}
+
+	function disable(/*object*/ obj, /*string*/ nodePart, /*boolean*/ disable){
+		obj.push({id: nodePart, attribute: "disabled", value: disable});
+	}
+
+	function display(/*object*/ obj, /*string*/ nodeDiv, /*string*/ display){
+		obj.push({id: nodeDiv, attribute: "display", value: display});
+	}	
```

Comment:
But where is the decision making for this table. Like which gets disabled when and which does not get disabled